Objects/longobject.c

```diff
@@ -770,6 +770,22 @@ _PyLong_Sign(PyObject *vv)
     return _PyLong_NonCompactSign(v);
 }
 
+int
+PyLong_Sign(PyObject *vv, int *sign)
+{
+    if (vv == NULL) {
+        PyErr_BadInternalCall();
+        return -1;
+    }
```

Comment:
I suggest to remove this code: the caller is responsible to make sure that the first parameter is not NULL. You can replace the check with an assertion: `assert(vv != NULL);` if you want.

Reply:
Well, there are commented out check(NULL) tests on many C-API functions, including PyLong_* ones.  My guess, the intention was to support that case.  @serhiy-storchaka ?

Reply:
I expect a comment `# CRASHES check(NULL)` in the unit test.

Reply:
Of course, but my point was isn't supposed that we should fix this in a new code?

Reply:
Checking for NULL makes the code slower. It's the responsibility of the caller to check for NULL.

Reply:
> Checking for NULL makes the code slower.

(I doubt it's  measurable.  Next check tests for subtypes of int...)

> It's the responsibility of the caller to check for NULL.

Ok, I'll read this just as a rule for new code.